lib/_http_server.js

```diff
@@ -582,10 +581,11 @@ function unconsume(parser, socket) {
   }
 }
 
+const SocketOn = net.Socket.prototype.on;
 function socketOnWrap(ev, fn) {
-  var res = net.Socket.prototype.on.call(this, ev, fn);
-  if (!this.parser) {
-    this.on = net.Socket.prototype.on;
+  const res = SocketOn.call(this, ev, fn);
+  if (this.parser === null) {
+    this.on = SocketOn;
```

Comment:
I wonder if we can `toFastProperties` on it here and if it would somehow help. Just curios not a nit or problem with the code.